src/app/lib/utilities/preprocessor/rules/cpsAssetPage/addHeadlineBlock/index.js

```diff
@@ -1,57 +1,57 @@
 import pathOr from 'ramda/src/pathOr';
+import deepClone from 'ramda/src/clone';
+import path from 'ramda/src/path';
+import {
+  getOffScreenHeadlineBlock,
+  getOnScreenHeadlineBlock,
+  getHeadlineBlock,
+} from './models';
 
-const addHeadingToSTY = jsonRaw => {
-  const headlineText = pathOr(
-    null,
-    ['promo', 'headlines', 'headline'],
-    jsonRaw,
-  );
+const getHeadlineBlocks = json => {
+  const headlineText = pathOr(null, ['promo', 'headlines', 'headline'], json);
 
   if (!headlineText) {
-    return jsonRaw;
+    return json;
   }
 
-  const newBlock = {
-    model: {
-      blocks: [
-        {
-          model: {
-            blocks: [
-              {
-                model: {
-                  blocks: [
-                    {
-                      model: {
-                        attributes: [],
-                        text: headlineText,
-                      },
-                      type: 'fragment',
-                    },
-                  ],
-                  text: headlineText,
-                },
-                type: 'paragraph',
-              },
-            ],
-          },
-          type: 'text',
-        },
-      ],
-    },
-    type: 'headline',
-  };
-
   return {
-    ...jsonRaw,
-    content: {
-      model: {
-        blocks: [
-          newBlock,
-          ...pathOr([], ['content', 'model', 'blocks'], jsonRaw),
-        ],
-      },
-    },
+    offScreenHeadlineBlock: getOffScreenHeadlineBlock(headlineText),
+    onScreenHeadlineBlock: getOnScreenHeadlineBlock(headlineText),
+    headlineBlock: getHeadlineBlock(headlineText),
   };
 };
 
-export default addHeadingToSTY;
+const firstBlockIsVideo = blocks => path(['0', 'type'], blocks) === 'video';
+
+const insertHeadlineBlocks = originalJson => {
+  const json = deepClone(originalJson);
+  const type = path(['metadata', 'type'], json);
+  const {
+    onScreenHeadlineBlock,
+    offScreenHeadlineBlock,
+    headlineBlock,
+  } = getHeadlineBlocks(json);
+  const blocks = path(['content', 'model', 'blocks'], json);
+
+  if (!blocks) {
+    return json;
+  }
+
+  if (type === 'MAP' && firstBlockIsVideo(blocks)) {
+    const videoBlock = blocks.slice(0, 1);
+    const mainBlocks = blocks.slice(1, blocks.length);
+
+    json.content.model.blocks = [
+      offScreenHeadlineBlock,
+      ...videoBlock,
+      onScreenHeadlineBlock,
+      ...mainBlocks,
+    ];
```

Comment:
This could all be simplified with `shift` - eg, delete lines 41-43, update line 46 to be `blocks.shift()`, update line 48 to be `...blocks`

Reply:
![](https://media.giphy.com/media/fs6rnt1K0YyVWRXNwY/giphy.gif)